### ISSUE TITLE
stripMargin type error with sbt 1.5.0, Scala 3.0.0-RC2 and scalatest 3.2.7

### ISSUE THREAD

**Issue by sim642** [scalatest/scalatest#2004]
I've been trying to migrate a Scala 2.13 project to Scala 3 and have been fighting a completely nonsensical type error for two days:
```
[error] 5 |  val exampleInput0: String = "foo".stripMargin('!')
[error]   |                              ^^^^^^^^^^^^^^^^^^^^^^
[error]   |                              Found:    Char
[error]   |                              Required: String
```

The only thing I've managed to discover is that this type error only occurs in a test class that `extends AnyFunSuite`. I'm not sure if scalatest is at fault here, but that's the only connection I have. The same test class compiles fine with Scala 3.0.0-RC2 and scalatest 3.2.7 on Scastie, so I don't know what the hell is happening.

Here's a minimal project where `Test / compile` in sbt outputs the above error: [scala3-stripMargin.zip](https://github.com/scalatest/scalatest/files/6330882/scala3-stripMargin.zip).

The [Scala 3 sbt migration tutorial](https://scalacenter.github.io/scala-3-migration-guide/docs/tutorials/sbt-migration.html#4-prepare-the-dependencies) recommended finding compatible versions on Scaladex's version matrix, but that shows some weird stuff for scalatest 3.2.7 and Scala 3.0.0-RC2: https://index.scala-lang.org/artifacts/scalatest/scalatest. Some packages like scalatest-funsuite don't have a green dot but sbt seems to download them just fine. Is scalatest 3.2.7 even supposed to work with Scala 3.0.0-RC2 or not?

**Comment by bvenners** [comment]
@sim642 Yes, 3.2.7 is built for 3.0.0-RC2: https://www.scalatest.org/release_notes/3.2.7

I'll have to study the scaladex matrix. Will return here and post once I understand better if any green dots are missing.

I was able to reproduce the problem, though for some reason I couldn't use the unzipped area. After copying the test class to a Scala 3 project I already had, I got:

```
[error] -- [E007] Type Mismatch Error: .../src/test/scala/StripMarginTest.scala:7:47 
[error] 7 |  val exampleInput0: String = "foo".stripMargin('!')
[error]   |                              ^^^^^^^^^^^^^^^^^^^^^^
[error]   |                              Found:    Char
[error]   |                              Required: String
[error]   |
[error]   |                              The following import might fix the problem:
[error]   |
[error]   |                                import org.scalactic.Prettifier.default
[error]   |
[error] -- [E007] Type Mismatch Error: .../src/test/scala/StripMarginTest.scala:16:24 
[error] 10 |    """|
[error] 11 |       !v
[error] 12 |       !|
[error] 13 |       !|
[error] 14 |       !|
[error] 15 |       !^
[error] 16 |       !|""".stripMargin('!')
[error]    |    ^
[error]    |    Found:    Char
[error]    |    Required: String
[error]    |
[error]    |    The following import might fix the problem:
[error]    |
[error]    |      import org.scalactic.Prettifier.default
[error]    |
[error] two errors found
[error] (Test / compileIncremental) Compilation failed
```

But the first line of code at least worked fine in the REPL. 

It is nonsensical, so you're not crazy! It might be a Dotty bug so we need to figure it out pronto. It doesn't make sense to me how anything ScalaTest does, which is extremely little by default, could cause that. I'll ask for help in the Dotty gitter channel, and see what they suggest.

Bill

**Comment by bvenners** [comment]
By the way, this is the test class in the zip file, and need be the only source file in the project:

```
import org.scalatest.funsuite.AnyFunSuite

class StripMarginTest extends AnyFunSuite {

  val exampleInput0: String = "foo".stripMargin('!')

  val exampleInput1: String =
    """|
       !v
       !|
       !|
       !|
       !^
       !|""".stripMargin('!')
}
```

**Comment by bvenners** [comment]
@sim642 Ok found the problem. It is this:

https://github.com/scalatest/scalatest/blob/4c7f2aafaaa150a00fd0a83e32e311668da5fbec/dotty/core/src/main/scala/org/scalatest/Assertions.scala#L1298

For some reason we have a `stripMargin` extension method. It maybe for handling strings in macros, but I'll find out. If we can delete it, then we will. If not, we need to add a sibling extension method that takes Char, because what is happening now is that our extension method that takes no args overpowers the one that takes `Char`. So then that returns `String`, and the `('!')` gets interpreted as `.apply('!'.toInt)`, which does have type `Char`. 

For you the workaround in the meantime is do a StringOps wrapper explicitly, like:

```
import org.scalatest.funsuite.AnyFunSuite
import scala.collection.StringOps

class StripMarginTest extends AnyFunSuite {

  val exampleInput0: String = StringOps("foo").stripMargin('!')

  val exampleInput1: String =
    StringOps(
      """|
         !v
         !|
         !|
         !|
         !^
         !|"""
    ).stripMargin('!')
  test("This test succeeded") { succeed }
  test("This test failed") { fail("I meant to do that!") }
}
```

**Comment by bvenners** [comment]
@cheeseng please add a test and fix for this issue and PR to 3.2.x-new. Tnx.

**Comment by cheeseng** [comment]
@sim642 Fyi this should be fixed by: 

https://github.com/scalatest/scalatest/pull/2005

and included in 3.2.9 build for Scala 3.0.0.
